debezium-core/src/main/java/io/debezium/pipeline/source/snapshot/incremental/AbstractIncrementalSnapshotContext.java

```diff
@@ -420,35 +433,35 @@ public boolean isEmpty() {
             return this.dataCollectionsToSnapshot.isEmpty();
         }
 
-        public boolean remove(List<DataCollection<T>> toRemove) {
+        public boolean remove(List<DataCollection<T>> toRemove, AbstractIncrementalSnapshotContext<T> context) {
             boolean removed = this.dataCollectionsToSnapshot.removeAll(toRemove);
-            this.dataCollectionsToSnapshotJson = computeJsonString();
+            this.dataCollectionsToSnapshotJson = computeJsonString(context);
             return removed;
         }
 
-        public String dataCollectionsAsJsonString() {
+        public String dataCollectionsAsJsonString(AbstractIncrementalSnapshotContext<T> context) {
 
             if (!Strings.isNullOrEmpty(dataCollectionsToSnapshotJson)) {
                 // A cached value to improve performance since this method is called in the "store"
                 // that is called during events processing
                 return dataCollectionsToSnapshotJson;
             }
 
-            return computeJsonString();
+            return computeJsonString(context);
         }
 
         public Queue<DataCollection<T>> getDataCollectionsToSnapshot() {
             return this.dataCollectionsToSnapshot;
         }
 
-        private String computeJsonString() {
+        private String computeJsonString(AbstractIncrementalSnapshotContext<T> context) {
```

Comment:
What do you think about passing the context as a constructor argument to `SnapshotDataCollection` instead, since the context in the first place owns the collection? That would avoid the need for passing `this` all over the various locations on method calls?